sycl/source/detail/event_impl.hpp

```diff
@@ -390,7 +393,7 @@ class event_impl {
   // If this event represents a submission to a
   // ur_exp_command_buffer_sync_point_t the sync point for that submission is
   // stored here.
-  ur_exp_command_buffer_sync_point_t MSyncPoint;
+  std::optional<ur_exp_command_buffer_sync_point_t> MSyncPoint;
```

Comment:
Can `MSyncPoint` be intentionally optional? If so, when?

Reply:
From what I understand, not every event has an associated MSyncPoint: E.g. if the event gets enqueued via `enqueueImpQueue` instead of `enqueueImpCommandBuffer`, the event created doesn't get any sync points.